src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```
@@ -265,11 +265,13 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
         <_AndroidApplicationSharedLibraryPath>$(IntermediateOutputPath)app_shared_libraries\</_AndroidApplicationSharedLibraryPath>
 	<_ResolvedUserAssembliesHashFile>$(IntermediateOutputPath)resolvedassemblies.hash</_ResolvedUserAssembliesHashFile>
 
+	<AndroidDexTool   Condition=" '$(AndroidDexTool)' != 'dx' ">d8</AndroidDexTool>
+	<_AndroidXA1025 Condition=" '$(EnableProguard)' == 'true' And '$(AndroidEnableProguard)' == '' And '$(AndroidDexTool)' == 'd8' And $(AndroidLinkTool) == '' ">true</_AndroidXA1025>
+	<_AndroidXA1026 Condition=" '$(AndroidEnableProguard)' == 'true' And '$(AndroidDexTool)' == 'd8' And $(AndroidLinkTool) == '' ">true</_AndroidXA1026>
 	<!-- $(EnableProguard) is an obsolete property that should be removed at some stage. -->
 	<AndroidEnableProguard Condition=" '$(AndroidEnableProguard)' == '' ">$(EnableProguard)</AndroidEnableProguard>
 	<AndroidEnableProguard Condition=" '$(AndroidLinkTool)' == 'proguard' ">True</AndroidEnableProguard>
 	<AndroidEnableMultiDex Condition=" '$(AndroidEnableMultiDex)' == '' ">False</AndroidEnableMultiDex>
-	<AndroidDexTool   Condition=" '$(AndroidDexTool)' != 'dx' ">d8</AndroidDexTool>
 	<AndroidDexTool   Condition=" '$(AndroidLinkTool)' == 'r8' ">d8</AndroidDexTool>
```

I know this isn't this PR, but maybe we should go ahead and change this line:
```suggestion
	<AndroidDexTool   Condition=" '$(AndroidLinkTool)' == 'r8' and '$(AndroidDexTool)' == 'dx' ">d8</AndroidDexTool>
```
I think this will prevent MSBuild from logging that the property changed from `d8` to `d8`.

---

Oh shoot. I forgot to take a look at adding this in the latest changes today. I'll plan to take a look the next time I'm working.

---

I got a chance to experiment with this today, and it looks like MSBuild has some built-in smarts about this.

Here's an excerpt of the diff of the diagnostic build output, comparing `$(AndroidDexTool)`=`dx` + `$(AndroidLinkTool)`=`r8` versus `$(AndroidDexTool)`=`d8` + `$(AndroidLinkTool)`=`r8`:

```diff
@@ -250,7 +250,6 @@ Property reassignment: $(ResolveNuGetPackageAssetsDependsOn)="
                _SetLatestTargetFrameworkVersion;
                ResolveProjectReferences
        " (previous value: "ResolveProjectReferences") at C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets (148,2)
-Property reassignment: $(AndroidDexTool)="d8" (previous value: "dx") at C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets (274,2)
 Property reassignment: $(AndroidEnableProguard)="True" (previous value: "") at C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets (279,2)
 Property reassignment: $(AssemblySearchPaths)="
                {HintPathFromItem};
```

That is, MSBuild logs a message when it changes the property value from `dx` to `d8`, but when the property value is already `d8` and MSBuild sets it set to `d8` again, it remains silent. I also see that this comment has now been marked as resolved. So I'll leave this `<AndroidDexTool/>` `Condition` as-is for now, but I'll be happy to adjust it if it's still a desired change for tidiness or similar.